doc/source/development/extending.rst

```diff
@@ -208,13 +208,201 @@ will
+if your ExtensionArray implements a compliant `__array_ufunc__` interface,
+when a numpy ufunc such as `np.floor` is invoked on your array, its
+implementation of `__array_ufunc__`  will bec called first and given the
+opportunity to compute the function. The return value needn't be a numpy
+ndarray (though it can be). In general, you want the return value to be an
+instance of your ExtensionArray. In some cases, your implementation can
+calculate the result itself (see for example TBD), or, if your ExtensionArray
+already has a numeric ndarray backing it, your implementation will itself
+invoke the numpy ufunc itself on it (see for example TBD). In either case,
+after computing the values, you will usually want to wrap the result as a new
+ExtensionArray instance and return it to the caller. Pandas will automatically
+use that Array as the backing ExtensionArray for a new Series object.
+
+.. note::
+    Before [NEP13](https://www.numpy.org/neps/nep-0013-ufunc-overrides.html),
+    numpy already provides a way of wrapping ufunc functions via `__array_prepare__`
+    and `__array_wrap__`, as documented in the Numpy documentation section
+    ["Subclassing Numpy"](http://docs.python.org/doc/numpy/user/basics.subclassing.html).
+    However, NEP13 seems to have largely superseded that mechanism.
+
+
+With ufuncs out of the way, we turn to the remaining numpy operations, such
+as `np.round`. The simplest way to support these operations is to simply
+implement a compatible method on your ExtensionArray. For example, if your
+ExtensionArray has a compatible `round` method on your ExtensionArray,
+When python involves `ser.round()`, :meth:``Series.round` will invoke
+`np.round(self.array)`, which will pass your ExtensionArray to the `np.round`
+method. Numpy will detect that your EA implements a compatible `round`
+and will invoke it to perform the operation. As in the ufunc case,
+your implementation will generally perform the calculation itself,
+or call numpy on its own backing numeric array, and in either case
+will wrap the result as a new instance of ExtensionArray and return that
+as a result. It is usually possible to write generic code to handle
+most ufuncs without having to provide a special case for each. For an example, see TBD.
+
+.. important::
+
+  When providing implementations of numpy functions such as `np.round`,
+  It essential that function signature is compatible with the numpy original.
+  Otherwise,, numpy will ignore it.
+
+  For example, the signature for `np.round` is `np.round(a, decimals=0, out=None)`.
+  if you implement a round function which omits the `out` keyword:
+
+.. code-block:: python
+
+    def round(self, decimals=0):
+        pass
+
+
+  numpy will ignore it. The following will work however:
+
+.. code-block:: python
+
+    def round(self, decimals=0, **kwds):
+        pass
+
+An alternative approach to implementing individual functions, is to override
```

Comment:
I wouldn't recommend this.

Reply:
I don't agree. When the EA is backed by a numeric array, It's actually the easiest and most generic solution.

With about 10 lines of getattr code, you can transparently proxy all (or most all) of the methods, by reinvoking numpy on your backing array, and casting the result. NEP18 is ornery, and implementing individual methods has
too much boilerplate. This is a viable and useful approach.